src/main/java/edu/si/fcrepo/AkubraIDConverter.java

```diff
@@ -0,0 +1,107 @@
+ * limited to any implied warranties of merchantability, fitness for a
+ * particular purpose, title or non-infringement; (2) do not assume any legal
+ * liability or responsibility for the accuracy, completeness, or usefulness of
+ * the software; (3) do not represent that use of the software would not
+ * infringe privately owned rights; (4) do not warrant that the software
+ * is error-free or will be maintained, supported, updated or enhanced;
+ * (5) will not be liable for any indirect, incidental, consequential special
+ * or punitive damages of any kind or nature, including but not limited to lost
+ * profits or loss of data, on any basis arising from contract, tort or
+ * otherwise, even if any of the parties has been warned of the possibility of
+ * such loss or damage.
+ *
+ * This distribution includes several third-party libraries, each with their own
+ * license terms. For a complete copy of all copyright and license terms, including
+ * those of third-party libraries, please see the product release notes.
+ */
+
+package edu.si.fcrepo;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+
+/**
+ * Utility functions extracted/modified based on org.fcrepo.server.storage.lowlevel.akubra.AkubraLowLevelStorage.
+ * Original code authored by Chris Wilper
+ *
+ * @author whikloj
+ */
+public class AkubraIDConverter {
+
+    /**
+     * Rather than bring in the entire PID class to get a string.
+     */
+    public static String FEDORA_URI = "info:fedora/";
+
+    /**
+     * Converts a token to a token-as-blobId.
+     *
+     * @param token the token to convert
+     * @return the AkubraLowLevelStorage BlobId
+     */
+    public static URI getBlobId(final URI token) {
+        return getBlobId(token.toString());
+    }
+
+    /**
+     * Converts a token to a token-as-blobId.
+     * <p>
+     * Object tokens are simply prepended with <code>info:fedora/</code>, whereas datastream tokens are additionally
+     * converted such that <code>ns:id+dsId+dsVersionId</code> becomes
+     * <code>info:fedora/ns:id/dsId/dsVersionId</code>, with the dsId and dsVersionId segments URI-percent-encoded
+     * with UTF-8 character encoding.
+     *
+     * @param token the token to convert.
+     * @return the blob id.
+     * @throws IllegalArgumentException if the token is not a well-formed pid or datastream token.
+     */
+    public static URI getBlobId(final String token) {
+        try {
+            final int i = token.indexOf('+');
+            if (i == -1) {
+                if (token.startsWith(FEDORA_URI)) {
+                    // We have the namespace so we're good.
+                    return new URI(token);
+                }
+                // Just prepend the namespace
+                return new URI(FEDORA_URI + token);
+            } else {
+                final String[] dsParts = token.substring(i + 1).split("\\+");
```

Comment:
Another in-line comment or two would be helpful here (through the end of this `else`). Under most circumstances it wouldn't matter, but this code is working with data produced by an EOL codebase the writers of which have long since moved on, so the more explicitness, the better.